packages/solid-urql/src/createSubscription.ts

```diff
@@ -0,0 +1,318 @@
+   * last `Operation` that the current state was for.
+   */
+  operation?: Operation<Data, Variables>;
+};
+
+/** Combines previous data with an incoming subscription result’s data.
+ *
+ * @remarks
+ * A `SubscriptionHandler` may be passed to {@link createSubscription} to
+ * aggregate subscription results into a combined {@link CreateSubscriptionState.data}
+ * value.
+ *
+ * This is useful when a subscription event delivers a single item, while
+ * you’d like to display a list of events.
+ *
+ * @example
+ * ```ts
+ * const NotificationsSubscription = gql`
+ *   subscription { newNotification { id, text } }
+ * `;
+ *
+ * const combineNotifications = (notifications = [], data) => {
+ *   return [...notifications, data.newNotification];
+ * };
+ *
+ * const [result, executeSubscription] = createSubscription(
+ *   { query: NotificationsSubscription },
+ *   combineNotifications,
+ * );
+ * ```
+ */
+export type SubscriptionHandler<T, R> = (prev: R | undefined, data: T) => R;
+
+/** Result tuple returned by the {@link createSubscription} hook.
+ *
+ * @remarks
+ * Similarly to a `createSignal` hook’s return value,
+ * the first element is the {@link createSubscription}’s state,
+ * a {@link CreateSubscriptionState} object,
+ * and the second is used to imperatively re-execute or start the subscription
+ * via a {@link CreateMutationExecute} function.
+ */
+export type CreateSubscriptionResult<
+  Data,
+  Variables extends AnyVariables = AnyVariables,
+> = [CreateSubscriptionState<Data, Variables>, CreateSubscriptionExecute];
+
+/** Hook to run a GraphQL subscription and get updated GraphQL results.
+ *
+ * @param args - a {@link CreateSubscriptionArgs} object, to pass a `query`, `variables`, and options.
+ * @param handler - optionally, a {@link SubscriptionHandler} function to combine multiple subscription results.
+ * @returns a {@link CreateSubscriptionResponse} tuple of a {@link CreateSubscriptionState} result,
+ * and an execute function.
+ *
+ * @remarks
+ * `createSubscription` allows GraphQL subscriptions to be defined and executed.
+ * Given {@link CreateSubscriptionArgs.query}, it executes the GraphQL subscription with the
+ * context’s {@link Client}.
+ *
+ * The returned result updates when the `Client` has new results
+ * for the subscription, and `data` is updated with the result’s data
+ * or with the `data` that a `handler` returns.
+ *
+ * @example
+ * ```ts
+ * import { gql, createSubscription } from '@urql/solid';
+ *
+ * const NotificationsSubscription = gql`
+ *   subscription { newNotification { id, text } }
+ * `;
+ *
+ * const combineNotifications = (notifications = [], data) => {
+ *   return [...notifications, data.newNotification];
+ * };
+ *
+ * const Notifications = () => {
+ *   const [result, executeSubscription] = createSubscription(
+ *     { query: NotificationsSubscription },
+ *     combineNotifications,
+ *   );
+ *   // ...
+ * };
+ * ```
+ */
+export const createSubscription = <
```

Comment:
Obvious TODO (mostly for my own benefit); This will have to match the `createQuery` implementation closely before merging

Reply:
Changed `createSubscription` to store source in state